core/src/data/clientGeoJsonSource.cpp

```diff
@@ -77,6 +78,83 @@ ClientGeoJsonSource::ClientGeoJsonSource(std::shared_ptr<Platform> _platform,
 
 ClientGeoJsonSource::~ClientGeoJsonSource() {}
 
+struct add_centroid {
+
+    geometry::point<double>& pt;
+
+    bool operator()(const geometry::polygon<double>& geom) {
+        if (geom.empty()) { return false; }
+        pt = centroid(geom.front().begin(), geom.front().end()-1);
+        if (isnan(pt.x) || isnan(pt.y)) {
+            return false;
+        }
+        return true;
+    }
+
+    bool operator()(const geometry::multi_polygon<double>& geom) {
+        float largestArea = 0.f;
+        size_t largestAreaIndex = 0;
+        auto size = geom.size();
+        for (size_t index = 0; index < geom.size(); index++) {
+            auto& g = geom[index];
+            if (g.empty()){
+                continue;
+            }
+            auto area = signedArea(g.front().begin(), g.front().end());
+            if (area > largestArea) {
+                largestArea = area;
+                largestAreaIndex = index;
+            }
+        }
+        return (*this)(geom[largestAreaIndex]);
```

Comment:
Ohh I only used this because that how it was being used with other piece of code here. But its one and the same thing :D.